src/components/Login/LoginOTPCard.tsx

```diff
@@ -69,11 +81,41 @@ export const LoginOTPCard: FunctionComponent<LoginOTPCard> = ({
     try {
       const response = await validateOTP(otp, mobileNumber, codeKey, endpoint);
       setIsLoading(false);
-      setAuthInfo(response.sessionToken, response.ttl.getTime(), endpoint);
-      navigation.navigate("CollectCustomerDetailsScreen");
+      const versionResponse = await getEnvVersion(
+        response.sessionToken,
+        endpoint
+      );
+
+      // Toggle between different environments
+      // using the DIST_ENV variable from features
+
+      // versionResponse.features.DIST_ENV
+      console.log(versionResponse.features.DIST_ENV);
+      if (
+        versionResponse.features.DIST_ENV === "VOUCHER" ||
+        versionResponse.features.DIST_ENV === "TOKEN"
+      ) {
+        setAuthInfo(response.sessionToken, response.ttl.getTime(), endpoint);
+        setFeatures(versionResponse.features);
+        setProducts(versionResponse.policies);
+
+        navigation.navigate("CollectCustomerDetailsScreen");
+      } else {
+        alert(
+          "Invalid Environment Error: Make sure you scanned a valid QR code"
+        );
+        // Reset to initial login state
+        setLoginStage("SCAN");
```

Comment:
I suggest we pass it a `resetStage` instead, so `LoginOTPCard` doesn't need to know about the `SCAN` state